components/gitpod-protocol/go/reconnecting-ws.go

```diff
@@ -27,10 +26,12 @@ type ReconnectingWebsocket struct {
 	closedCh chan struct{}
 	connCh   chan chan *websocket.Conn
 	errCh    chan error
+
+	log *logrus.Entry
```

Comment:
Just for clarification: What was the motivation to introduce this pattern? And: IMO `Entry` does represent a single line. Does this result in a "last log wins" situation?

Reply:
really! then I should pass `Logger`?

Reply:
https://github.com/sirupsen/logrus/blob/master/entry.go#L44

Reply:
I switched to `Logger`, but i'm not sure that `Entry` actually represents a line, it seems to be more a builder pattern, i.e. you can create a subentry with some field and then use it to produce many lines with different messages.